lib/mox.ex

```diff
@@ -558,4 +563,12 @@ defmodule Mox do
 
   defp times(1), do: "once"
   defp times(n), do: "#{n} times"
+
+  # Find the pid of the actual caller
+  defp caller_pid do
+    case Process.get(:"$callers") do
+      nil -> self()
+      pids when is_list(pids) -> List.last(pids)
```

Comment:
I think we should actually pass a list to the server and start looking up from head to tail. :)